metagraph/src/annotation/binary_matrix/multi_brwt/brwt.cpp

```diff
@@ -57,6 +58,40 @@ BRWT::SetBitPositions BRWT::get_row(Row row) const {
     return row_set_bits;
 }
 
+Vector<std::pair<BRWT::Column, uint64_t>> BRWT::get_column_ranks(Row i) const {
+    assert(i < num_rows());
+
+    // check if the row is empty
+    uint64_t rank = nonzero_rows_->conditional_rank1(i);
+    if (!rank)
+        return {};
+
+    Vector<std::pair<BRWT::Column, uint64_t>> row;
+
+    // check whether it is a leaf
+    if (!child_nodes_.size()) {
+        assert(assignments_.size() == 1);
+
+        // the bit is set
+        for (size_t j = 0; j < assignments_.size(); ++j) {
+            row.emplace_back(j, rank);
```

Comment:
is `row` every resized later? if not, reserve space

Reply:
I don't know it's size. Setting it to num_columns() will hugely overestimate it in most of the cases.
So without the following shrink_to_fit this may accumulate and take in total tonnes of memory.

Also, I think it won't make a difference in the runtime anyway.

Reply:
For this base case, we know that `row`'s final size is gonna be `assignments_.size()`.

I'm not really concerned about runtime, it's that it might be less likely to use the small vector optimizations if the vector is not pre-allocated

Reply:
> For this base case, we know that row's final size is gonna be assignments_.size()

... which is equal to `num_columns()`

Reply:
actually, from the assertion, we know that it's final size is gonna be 1, so we can even just return `{ { 0, rank } }`

Reply:
Ok, this was written for a more general case, but yeah, currently there can only be a single column in leaves.

Though, I am sure this will have a zero effect on the memory or CPU usage...